test-lib.sh

```diff
@@ -246,19 +246,32 @@ function ct_build_s2i_npm_variables()
 function ct_npm_works() {
   local tmpdir=$(mktemp -d)
   : "  Testing npm in the container image"
+  cid_file="${tmpdir}/cid"
   docker run --rm ${IMAGE_NAME} /bin/bash -c "npm --version" >${tmpdir}/version
   if [ $? -ne 0 ] ; then
     echo "ERROR: 'npm --version' does not work inside the image ${IMAGE_NAME}." >&2
     return 1
   fi
 
-  docker run $(ct_mount_ca_file) --rm ${IMAGE_NAME} /bin/bash -c "npm install jquery && test -f node_modules/jquery/src/jquery.js"
+  docker run --user=100001 $(ct_mount_ca_file) --rm --cidfile=${cid_file} ${IMAGE_NAME}-testapp &
```

Comment:
I expect this is the culprit for the hang. You run the container in the background (why not run it detached instead?) but you never kill it.

Reply:
Killed the container manually and the job completed

Reply:
I have had in the commit also cleaning the container https://github.com/sclorg/container-common-scripts/pull/129#discussion_r352220415 but delete it.

I was inspired by s2i-nodejs-container. 
https://github.com/sclorg/s2i-nodejs-container/blob/master/test/run#L292

But missing parts are:
* https://github.com/sclorg/s2i-nodejs-container/blob/master/test/run#L92
* https://github.com/sclorg/s2i-nodejs-container/blob/master/test/run#L97

My thoughts were, that once the job is finished, then the container is automatically destroyed.

Running container in detached mode can be another way.